src/components/Infra/InfraListPage.tsx

```diff
@@ -49,23 +84,54 @@ const InfraListPage: React.FC<InfraListPageProps> = ({ namespace }) => {
     namespace,
   });
 
+  const columns: TableColumn<CIM.InfraEnvK8sResource>[] = React.useMemo(() => {
+    const cols = [
+      {
+        title: 'Name',
+        id: COL_NAME,
+        transforms: [sortable],
+        sort: 'metadata.name',
+      },
+    ];
+
+    if (!namespace) {
+      cols.push({
+        title: 'Project',
+        id: COL_PROJECT,
+        transforms: [sortable],
+        sort: 'metadata.namespace',
+      });
+    }
+
+    cols.push({
+      title: 'Location',
+      id: COL_LOCATION,
+      transforms: [sortable],
+      sort: `metadata.labels['${AGENT_LOCATION_LABEL_KEY}']`,
+    });
+
+    cols.push({
+      title: 'Hosts',
+      id: COL_AGENTS,
+      transforms: [],
+      sort: '',
+    });
+
+    return cols.filter(Boolean);
+  }, [namespace]);
+
   return (
     <>
       <ListPageHeader title="Infrastructures">
-        <ListPageCreate
-          groupVersionKind={InfraEnvKind}
-          // namespace={namespace || 'assisted-installer'}
-        >
-          Create
-        </ListPageCreate>
+        <ListPageCreate groupVersionKind={InfraEnvKind}>Create</ListPageCreate>
       </ListPageHeader>
       <ListPageBody>
         <VirtualizedTable
           loaded={loaded}
           loadError={loadError}
           data={infras}
           unfilteredData={infras /* So far we do not have filters */}
-          Row={InfraRow}
+          Row={(props) => <InfraRow isNamespaced={!!namespace} {...props} />}
```

Comment:
this will break Virtualized table performance.

A correct approach to hide namespace column (that is column with `namespace` id so you need to set `COL_PROJECT = 'namespace'`) is to use `useActiveColumns` hook (its already exposed via SDK) which does that automatically.

See https://github.com/openshift/console/blob/master/frontend/public/components/pod.tsx#L838-L842 as example. Only the first param in `useActiveColumns` is required (which is what we need)

Reply:
Thanks, done